Tests/MapboxNavigationTests/NavigationViewControllerTests.swift

```diff
@@ -485,6 +485,46 @@ class NavigationViewControllerTests: TestCase {
                        floatingButton,
                        "Unexpected floating button.")
     }
+    
+    func testShieldStyleWithNavigationMapViewInjection() {
+        let nightStyleURI: StyleURI = .navigationNight
+        let dayStyleURI: StyleURI = .navigationDay
+        
+        let spriteRepository = SpriteRepository.shared
+        let config = URLSessionConfiguration.default
+        config.timeoutIntervalForRequest = 1.0
+        spriteRepository.sessionConfiguration = config
+        
+        let styleLoadedExpectation = XCTestExpectation(description: "Style updated expectation.")
+        spriteRepository.updateStyle(styleURI: nightStyleURI) { _ in
+            styleLoadedExpectation.fulfill()
+        }
+        wait(for: [styleLoadedExpectation], timeout: 2.0)
+        XCTAssertEqual(spriteRepository.userInterfaceIdiomStyles[.phone],
+                       nightStyleURI,
+                       "Failed to update the style of SpriteRepository singleton to Night style.")
+        
+        class CustomNavigationMapView: NavigationMapView { }
+        let injected = CustomNavigationMapView()
+        injected.mapView.mapboxMap.style.uri = dayStyleURI
+        
+        let navService = MapboxNavigationService(indexedRouteResponse: initialRouteResponse,
+                                                 customRoutingProvider: nil,
+                                                 credentials: Fixture.credentials)
+        let navOptions = NavigationOptions(styles: [DayStyle()],
+                                           navigationService: navService,
+                                           navigationMapView: injected)
+        let subject = NavigationViewController(for: initialRouteResponse, navigationOptions: navOptions)
```

Comment:
In https://github.com/mapbox/mapbox-navigation-ios/pull/4188 some changes were added for case when `NavigationMapView` injection is done directly by accessing `NavigationViewController.navigationMapView` property. It'd be also great to test such case. It can be added in scope of different PR, not to block this PR.